inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,376 @@
+        :return: returns the new number of conditions passed,
+         the new key for this recursion and if a condition got checked and the failed the check
+        """
+        key = self.keypath[position]
+        condition_failed = False
+        if self.conditions:
+            for condition in self.conditions:
+                should_check = False
+                if position < len(condition.get('keypath')):
+                    if key != condition.get('keypath')[position] and\
+                     (position == 0 or self.keypath[position - 1] == condition.get('keypath')[position - 1]):
+                        should_check = True
+                    elif key == condition.get('keypath')[position] and position == len(self.keypath) - 1:
+                        should_check = True
+                    if should_check:
+                        if not check_value(record=record, schema=schema, keypath=condition.get('keypath'),
+                                           update_value=condition.get('value', ''),
+                                           match_type=condition.get('match_type', ''), position=position):
+                            condition_failed = True
+                        else:
+                            conditions_passed = conditions_passed + 1  # number of conditions that passed successfully
+        return conditions_passed, key, condition_failed
+
+
+class Addition(Action):
+    """Class that when applied adds a new primitive field or object to the selected path
+    if the conditions are satisfied"""
+    def apply(self, record, schema, position=0, conditions_passed=0):
+        """
+        :param record: record in which the field to be updated resides
+        :param schema: schema on which our record is based
+        :param position: level of recursion
+        :param conditions_passed: number of conditions that run and passed
+        :return:
+        """
+        conditions_passed, key, condition_failed = self.get_next_key_and_condition_check(record, schema, position, conditions_passed)
+        if not schema or condition_failed:
+            return
+        if not record.get(key):
+            if self.conditions and conditions_passed < len(self.conditions):
+                return
+            remaining_path = self.keypath[position:]
+            record.update(create_schema_record(schema, remaining_path, self.value))
+            self.changed = True
+            return
+        if position == len(self.keypath) - 1:
+            if isinstance(record[key], list):
+                record[key].append(self.value)
+                self.changed = True
+        else:
+            new_schema = schema_progression(schema, key)
+            if isinstance(record[key], list):
+                for array_record in record[key]:
+                    self.apply(array_record, new_schema, position + 1, conditions_passed)
+            else:
+                self.apply(record[key], new_schema, position + 1, conditions_passed)
```

Comment:
There are no empty lists in our records, so if the case is that the list is not existing, it's key is going to be missing so im going to create it. :)

Reply:
That is something that might change in the future (not having empty lists), and it does not hurt now to make sure here to be able to handle those too, so better be on the safe side (it will be a mess to debug later in case we add empty lists).